server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -104,11 +107,30 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
-    private UDTValue convertToUDTValue(Optional<ZonedDateTime> zonedDateTimeOptional) {
+    private Insert createSpecificUpdate(VacationPatch vacationPatch, Insert baseInsert) {
+        return insertGeneratorPipeline
+            .apply(vacationPatch)
+            .apply(baseInsert);
+    }
+
+    public <T> Function<Insert, Insert> addValue(String field, MissableOptional<T> missableValue) {
```

Comment:
function name `addValue` doesn't express what it does.
You are returning a function that either add an update clause to a querybuilder or do nothing.
What do you think of `applyPatchForField` ?